### PR TITLE
fix: Fix error where a lot of domains were not detected

### PR DESCRIPTION
**Pull request by jpstevens**
A number of domains were not recognised, due to issues with the separator logic in the trie serializer.

TLDs which didn't work, but now do, are as follows:

### niteroi.br / *.nom.br
Changes trie snapshot from `niteroinom>*` to `niteroi,nom>*`

### police.uk / *.sch.uk
Changes trie snapshot from `policesch>*` to `police,sch>*`

### budejju.no / nes.buskerud.no
Changes trie snapshot from `budejjubuskerud>nes` to `budejju,buskerud>nes`

### hønefoss.no / os.hordaland.no
Changes trie snapshot from `hønefosshordaland>os` to `hønefoss,hordaland>os`

### molde.no / heroy.more-og-romsdal.no
Changes trie snapshot from `moldemore-og-romsdal>heroy` to `molde,more-og-romsdal>heroy`

### nordkapp.no / bo.nordland.no
Changes trie snapshot from `nordkappnordland>bo` to `nordkapp,nordland>bo`

### osterøy.no / valer.ostfold.no
Changes trie snapshot from `osterøyostfold>valer` to `osterøy,ostfold>valer`

### tananger.no / bo.telemark.no
Changes trie snapshot from `tanangertelemark>bo` to `tananger,telemark>bo`

### vestby.no / sande.vestfold.no
Changes trie snapshot from `vestbyvestfold>sande` to `vestby,vestfold>sande`

### haugesund.no / os.hedmark.no
Changes trie snapshot from `haugesundhedmark>os` to `haugesund,hedmark>os`

### CODE REVIEW

**Comment by jpstevens** [comment]
Fixes #64

**Comment by glasser** [comment]
It's a lot more than just the ones that are listed there.  Notably this fixes s3.amazonaws.com too.

**Comment by jhnns** [comment]
Thanks for your PR and feedback. I'll review it the next days.

**Comment by jpstevens** [comment]
Hey @jhnns - can you please merge and release these changes?

**Comment by flotwig** [comment]
@jhnns bump, this would be great to see fixed

File: src/tries/serializeTrie.js [hunk 1]
```diff
@@ -42,14 +42,15 @@ function lineToString(line, i, arr) {
...
+
         if (indexOfDifference === -1) {
             // Identical lines
             return "";
         }
         if (indexOfDifference === 0) {
             // line and prevLine are completely different
             separatorFromPrev = SEPARATORS.RESET;
-        } else if (prevLine.length === line.length && indexOfDifference === line.length - 1) {
+        } else if (prevLine.length <= line.length && indexOfDifference === prevLine.length - 1) {
```
Comment (jhnns) [inline src/tries/serializeTrie.js]: I think the whole `length` check is not necessary or too strict. We should just compare `indexOfDifference === prevLine.length - 1)

**Comment by jhnns** [comment]
Awesome 🎉 🎉 🎉 thank you so much for this PR. This fixes a lot of domains.

**Comment by jhnns** [comment]
Published with `v2.1.8` 🚀

**Pull request merged**